identity-provider-service/src/bin/main.rs

```diff
@@ -819,27 +829,70 @@ async fn main() -> anyhow::Result<()> {
     // verified attributes.
     let create_identity_v1 = warp::get()
         .and(warp::path!("api" / "v1" / "identity" / "create" / String))
-        .and_then(move |id_cred_pub_hash: String| {
+        .and(warp::query::<WithExpiry>())
+        .and_then(move |id_cred_pub_hash: String, parameters: WithExpiry| {
             create_signed_identity_object_v1(
                 Arc::clone(&server_config_create_v1),
                 Arc::clone(&create_db_v1),
                 client_v1.clone(),
                 id_cred_pub_hash,
+                parameters.expiry,
             )
         });
 
+    // Endpoint for creating failed identities. The identity verification service
+    // will forward the user to this endpoint after they have created a list of
+    // verified attributes, and chosen to fail the identity.
+    // N.B. This is used for testing
+    let fail_identity = warp::get()
+        .and(warp::path!("api" / String / "identity" / "fail" / String))
+        .and(warp::query())
+        .and_then(
+            move |version: String, id_cred_pub_hash: String, query: HashMap<String, String>| {
+                let delay = query
+                    .get("delay")
+                    .map(|d| d.parse::<u64>().ok())
+                    .flatten()
+                    .unwrap_or(10);
```

Comment:
I'd prefer if we instead maybe logged errors instead of just `unwrap`'ing them. It's easier to debug.